artemis/modules/dangling_dns_detector.py

```diff
@@ -220,13 +298,24 @@ def run(self, current_task: Task) -> None:
             return
 
         result: list[dict[str, Any]] = []
-        self.check_dns_ip_records_are_alive(domain, result)
-        self.check_cname(domain, result)
-        self.check_ns(domain, result)
+
+        last_ip_scan = retry_count >= Config.Modules.DanglingDnsDetector.DANGLING_DNS_NUMBER_OF_RETRIES_FOR_IP
+        ip_records_alive = self.check_dns_ip_records_are_alive(domain, result, save_results=last_ip_scan)
+
+        if task_type == TaskType.DOMAIN_THAT_MAY_NOT_EXIST:
+            # we only check that once, no need to rerun that when suspecting dangling ip
+            self.check_cname(domain, result)
+            self.check_ns(domain, result)
+
+        retry_message = None
+        if self.handle_scheduling_retry(domain, current_task, ip_records_alive, last_ip_scan):
+            retry_message = f"Defined domain has dangling IP record, scheduling {retry_count + 1} retry."
```

Comment:
shouldn't we save the message only after the **last** check?

Reply:
Yeah, I wasnt sure if we should save that. The reasoning was, to be able to quickly check in the dashboard how many retries there were. But we can save only after last one, I'm fine with both.